### PR TITLE
Switch to codingwell/scala-guice and refactor Guice code

### PR DESCRIPTION
**Pull request by gmethvin**
This guice extension library seems a bit more comprehensive than the one we were using. It supports multibindings to scala sets and has its own version of our RichInjector called ScalaInjector.

### CODE REVIEW

File: shoebox/project/Build.scala [hunk 1]
```diff
@@ -61,7 +61,8 @@ object ApplicationBuild extends Build {
       "com.newrelic.agent.java" % "newrelic-agent" % "2.18.0",
       "com.google.inject" % "guice" % "3.0",
       "com.google.inject.extensions" % "guice-multibindings" % "3.0",
-      "com.tzavellas" % "sse-guice" % "0.7.1"
+      "net.codingwell" % "scala-guice_2.10" % "3.0.2",
+      "net.codingwell" %% "scala-guice" % "3.0.2"
```
Comment (andrewconner) [inline shoebox/project/Build.scala]: Are there two libraries?

**Comment by eishay** [comment]
nice!

File: shoebox/project/Build.scala (hunk 1 above)
Comment (gmethvin) [inline shoebox/project/Build.scala]: This is just what it gave for sbt in the readme. Maybe we don't need the second one? It doesn't show it for maven.

Comment (gmethvin) [inline shoebox/project/Build.scala]: Yeah, seems like that's the case. I'll remove the second library.

Comment (andrewconner) [inline shoebox/project/Build.scala]: Can you remove the first instead? It's more future proof, since it doesn't hard code Scala 2.10

**Pull request merged**